File: serviceAPI/VaccineAPI/Models/Vaccine.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace VaccineAPI.Models
{
    public class Vaccine
    {
        [Key]
        [Column("id")]
        public long Id { get; set; }

        [Required]
        [Column("name")]
        public string Name { get; set; } = string.Empty;

        [Column("manufacturer")]
        public string? Manufacturer { get; set; }

        [Column("expiration_date")]
        public DateTime? ExpirationDate { get; set; }

        [Column("quantity")]
        public int Quantity { get; set; }

        [Column("description")]
        public string? Description { get; set; }

        [JsonIgnore]
        public ICollection<VaccineDetail>? VaccineDetails { get; set; }
    }
}

File: frontend/Vaccine/src/app/model/vaccine.ts
import { VaccineDetail } from "./vaccine-detail";

export interface Vaccine {
    id?: number; // Id có thể là undefined khi thêm mới
    name: string;
    manufacturer: string;
    expirationDate: string;
    quantity: number;
    description: string;
    details: VaccineDetail[]; // Mảng các vaccine details
  }

  